cli/src/main.rs

```diff
@@ -1,8 +1,89 @@
+use db::PgPool;
+use tokio;
+use clap::{Parser, Subcommand};
+use db::sqlx;
+use std::fs;
+use std::io::Write;
+use uuid::Uuid; // Ensure you have the `uuid` crate in your Cargo.toml
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct KnowledgeArgs {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    Login,
+    Signup,
+    DisplayUsers,
+}
+
 #[tokio::main]
 async fn main() -> color_eyre::Result<()> {
-    let _db_pool = db::setup_db_pool().await?;
+    let args = KnowledgeArgs::parse();
+    let db_pool = db::setup_db_pool().await?;
+
+    match args.command {
+        Command::Login => {
+            println!("Login not implemented yet")
+        },
+        Command::Signup => {
+            sign_up(&db_pool).await?;
+        },
+        Command::DisplayUsers => {
+            display_users(&db_pool).await?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn persist_auth_session(user_id: Uuid) -> color_eyre::Result<()> {
+    if fs::metadata("auth.txt").is_ok() {
```

Comment:
So this definitely works! But we can be a bit more specific with this check likely!

I think we might want to use either `Path::exists` or maybe `Path::try_exists`!

These will _just_ check that the file exists, and doesn't grab the metadata.

https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists
https://doc.rust-lang.org/std/path/struct.Path.html#method.exists

The difference between the two is that `try_exists` will return an `Err` is there are errors checking the file system, while  `exists` will simply return `false`
This allows `try_exists` to be able to tell the difference between an error and the file existing!
But I don't think that we really care about the difference between those two cases here, so I think `exists` is likely fine!